src/main/java/com/greghaskins/spectrum/ParamaterizedSyntax.java

```diff
@@ -0,0 +1,215 @@
+import com.greghaskins.spectrum.internal.parameterized.ParameterizedDefinitionBlock.SixArgBlock;
+import com.greghaskins.spectrum.internal.parameterized.ParameterizedDefinitionBlock.ThreeArgBlock;
+import com.greghaskins.spectrum.internal.parameterized.ParameterizedDefinitionBlock.TwoArgBlock;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+/**
+ * Common syntax for parameterization.
+ */
+public interface ParamaterizedSyntax {
+  /**
+   * Describes a parameterized specification suite. E.g.
+   * <pre><code>
+   *         describe("Cuke eating without gherkin",
+               (start, eat, left) -&gt; {
+                 it("eats cukes properly", () -&gt; {
+                   CukeEater me = new CukeEater(start);
+                   me.eatCucumbers(eat);
+                   assertThat(me.remainingCucumbers(), is(left));
+                 });
+               },
+  
+               withExamples(
+                 example(12, 5, 7),
+                 example(20, 5, 15))
+               );
+   * </code></pre>
+   * @param name name of the parameterized test
+   * @param block a {@link ParameterizedDefinitionBlock} to execute that consumes the parameters
+   *              from the examples
+   * @param examples the examples to run through, built using
+   *                {@link ParamaterizedSyntax#withExamples(Example[])}
+   * @param <T> the type parameter, best derived implicitly from the examples
+   */
+  static <T> void describeParameterized(final String name, final T block,
+      final Stream<Example<T>> examples) {
+    Spectrum.describe(name, () -> {
+      Spectrum.context("Examples:", () -> {
+        examples.forEach(example -> {
+          Spectrum.describe(example.toString(), () -> example.runDeclaration(block));
+        });
+      });
+    });
+  }
+
+
+  /**
+   * Construct examples. Used to convert from individual objects created with
+   * {@link #example(Object)} type methods into a type-implicit stream. You should try to lay out
+   * your examples like a data table as that's what they essentially are. Better than just
+   * providing some primitives in an example block would be to provide some objects with fields
+   * that represent the input parameters more strongly. However, this pattern allows you to create
+   * ad-hoc tuples of type-consistent columns with rows of object values.
+   * @param examples example objects - use the 1-8 parameter versions of {@link #example(Object)}
+   * @param <T> the resulting number-of-arguments type
+   * @return a new stream of examples for parameterized tests to use
+   */
+  @SafeVarargs
+  static <T> Stream<Example<T>> withExamples(Example<T>... examples) {
+    return Arrays.stream(examples);
+  }
+
+  /**
+   * Single value example.
+   * @param t1 the parameter
+   * @param <T1> the type of param 1
+   * @return single value example
+   */
+  static <T1> Example<OneArgBlock<T1>> example(T1 t1) {
```

Comment:
`t1`, `t2`, and `t3` don't feel natural to me as parameter names. The convention is usually `arg0`, `arg1`, etc. for generic unnamed parameters, but I wonder if `param1`, `param2`, etc. would be better. Thoughts?

For the one-arg case, it should just be `<T>` since there's only one type.